cmp/grid/GridModel.js

```diff
@@ -1085,9 +1103,26 @@ export class GridModel {
         };
     }
 
+    parseChooserModel(chooserModel) {
+        if (XH.isMobileApp) {
+            return chooserModel ? this.markManaged(new MobileColChooserModel(this)) : null;
+        }
+
+        if (chooserModel instanceof ColChooserModel) {
+            return new DesktopColChooserModel(chooserModel);
+        }
+
+        if (isPlainObject(chooserModel)) {
+            return this.markManaged(new DesktopColChooserModel(defaults(chooserModel,
+                {gridModel: this})));
+        }
 
-    createChooserModel() {
-        return XH.isMobileApp ? new MobileColChooserModel(this) : new DesktopColChooserModel(this);
+        if (isBoolean(chooserModel)) {
+            return chooserModel ? this.markManaged(new DesktopColChooserModel({gridModel: this})) : null;
+        }
+
+        console.warn(`colChooserModel config in GridModel not supported: ${chooserModel}`);
```

Comment:
I can go either way on the final step to validate that it's a boolean - would also be fine just using truthiness at this point - i.e. after the check for plainObject simply `return chooserModel ? this.markManaged(new DesktopColChooserModel({gridModel: this})) : null;`

But if we are going to check for the boolean we should throw instead of console.warning.  That's a dev-time error and there's no need to be subtle about it.

Reply:
ok -- will take your first suggestion and remove the console warning